test/users.js

```diff
@@ -728,3 +730,576 @@ describe('/users GET', () => {
+      .catch(err => {
+        const res = err.response;
+        res.should.have.status(401);
+        res.should.be.json;
+
+        expect(res.body).to.equal('Invalid authorization token data');
+        done();
+      })
+      .catch(err => done(err));
+  });
+
+  it('should fail if the token does not contain an email without calling the database', done => {
+    const invalidToken = jwt.encode({ id: 1 }, config.common.session.secret);
+    const validUser = {
+      firstName: 'John',
+      lastName: 'Doe',
+      email: 'john.doe@wolox.com.ar',
+      password: '12345678'
+    };
+
+    chai
+      .request(server)
+      .post('/users/admin')
+      .set(config.common.session.header_name, invalidToken)
+      .send(validUser)
+      .catch(err => {
+        const res = err.response;
+        res.should.have.status(401);
+        res.should.be.json;
+
+        expect(res.body).to.equal('Invalid authorization token');
+        done();
+      })
+      .catch(err => done(err));
+  });
+
+  it('should fail if the token does not contain an id without calling the database', done => {
+    const invalidToken = jwt.encode({ email: 'jane.doe@wolox.cl' }, config.common.session.secret);
+    const validUser = {
+      firstName: 'John',
+      lastName: 'Doe',
+      email: 'john.doe@wolox.com.ar',
+      password: '12345678'
+    };
+
+    chai
+      .request(server)
+      .post('/users/admin')
+      .set(config.common.session.header_name, invalidToken)
+      .send(validUser)
+      .catch(err => {
+        const res = err.response;
+        res.should.have.status(401);
+        res.should.be.json;
+
+        expect(res.body).to.equal('Invalid authorization token');
+        done();
+      })
+      .catch(err => done(err));
+  });
+
+  it('should fail if the token is invalid', done => {
+    const invalidToken = 'blahblah';
+    const validUser = {
+      firstName: 'John',
+      lastName: 'Doe',
+      email: 'john.doe@wolox.com.ar',
+      password: '12345678'
+    };
+
+    chai
+      .request(server)
+      .post('/users/admin')
+      .set(config.common.session.header_name, invalidToken)
+      .send(validUser)
+      .catch(err => {
+        const res = err.response;
+        res.should.have.status(401);
+        res.should.be.json;
+
+        expect(res.body).to.equal('Invalid authorization token');
+        done();
+      })
+      .catch(err => done(err));
+  });
+
+  it('should fail if the token is not in the correct header key', done => {
+    const validToken = jwt.encode({ id: 1, email: 'jane.doe@wolox.cl' }, config.common.session.secret);
+    const validUser = {
+      firstName: 'John',
+      lastName: 'Doe',
+      email: 'john.doe@wolox.com.ar',
+      password: '12345678'
+    };
+
+    chai
+      .request(server)
+      .post('/users/admin')
+      .set('x-access-token', validToken)
+      .send(validUser)
+      .catch(err => {
+        const res = err.response;
+        res.should.have.status(401);
```

Comment:
big refactor: let's use just expect for everything. You can do it in another branch

Reply:
No problem, it's just a RegExp.